service/worker/migration/activities.go

```diff
@@ -383,20 +388,42 @@ func (a *activities) SeedReplicationQueueWithUserDataEntries(ctx context.Context
 			return err
 		}
 
-		request := &matchingservice.SeedReplicationQueueWithUserDataEntriesRequest{
-			NamespaceId:   describeResponse.GetNamespaceInfo().Id,
-			NextPageToken: heartbeatDetails.nextPageToken,
-			PageSize:      int32(params.PageSize),
+		request := &persistence.ListTaskQueueUserDataEntriesRequest{
+			NamespaceID:   describeResponse.GetNamespaceInfo().Id,
+			NextPageToken: heartbeatDetails.NextPageToken,
+			PageSize:      params.PageSize,
 		}
-		response, err := a.matchingClient.SeedReplicationQueueWithUserDataEntries(ctx, request)
+		response, err := a.taskManager.ListTaskQueueUserDataEntries(ctx, request)
 		if err != nil {
-			a.logger.Error("Seed task queue user data failed", tag.WorkflowNamespaceID(request.NamespaceId), tag.Error(err))
+			a.logger.Error("List task queue user data failed", tag.WorkflowNamespaceID(request.NamespaceID), tag.Error(err))
 			return err
 		}
+		for idx, entry := range response.Entries {
+			if heartbeatDetails.IndexInPage > idx {
+				continue
+			}
+			heartbeatDetails.IndexInPage = idx
+			activity.RecordHeartbeat(ctx, heartbeatDetails)
```

Comment:
how big is a page? heartbeating once per page seems like enough? it's harmless of course